torchvision/datasets/coco.py

```diff
@@ -30,7 +30,11 @@ def __init__(
         super().__init__(root, transforms, transform, target_transform)
         from pycocotools.coco import COCO
 
-        self.coco = COCO(annFile)
+        try:
+            self.coco = COCO(annFile)
+        except FileNotFoundError as error:
+            raise RuntimeError(f"The file {annFile} does not exist or is corrupt.") from error
```

Comment:
I'm not sure we should be changing the call-sites so that the current test infra works. Plus, `FileNotFoundError` is a meaningful type of error to be raised in here. Would your plan be to change all locations in the code to always raise `RuntimeError`?

Reply:
> I'm not sure we should be changing the call-sites so that the current test infra works.

IMO every dataset should handle corrupted or non-existent files properly. I wouldn't do that to satisfy our tests, but rather because it can make the life of the user easier with a descriptive error message. Plus, this is what you agreed to

https://github.com/pytorch/vision/blob/67681a7647382e6a87034b749e4379f1ec7749b8/test/datasets_utils.py#L398-L401

> Plus, `FileNotFoundError` is a meaningful type of error to be raised in here.

I agree. I'll update the test infrastructure.

> Would your plan be to change all locations in the code to always raise `RuntimeError`?

That was the original plan, yes. Maybe we can differentiate between `RuntimeError` for corrupted and `FileNotFoundError` for non-existent files. If we want to make sure we could also opt to create an `FileCorruptedError(Exception)` and check for that instead of a more general `RuntimeError`.